### PR TITLE
Ratio math documentation, some improvement of Price Authority API doc

### PR DESCRIPTION
**Pull request by tyg**

### CODE REVIEW

File: main/zoe/api/contract-support/price-authority.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# PriceAuthority Methods
+
+We rely on `PriceAuthority` oracles. A `PriceAuthority`
+gives reliable quotes for prices. The quotes might be based on broad surveys
+of prices across the ecosystem, or might come directly from an AMM (**tyg todo: What's AMM stand for?)** like our
```
Comment (katelynsills) [inline main/zoe/api/contract-support/price-authority.md:5]: automatic market maker

File: main/zoe/api/contract-support/price-authority.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+## `makeQuoteNotifier(amountIn, brandOut)`
+- `amountIn`: `{ Amount }`
+- `brandOut`: `{ Brand }`
+- Returns: `{ ERef<Notifier<PriceQuote>> }`
+- Be notified of the latest `PriceQuotes` for a given `amountIn`. The issuing
+  rate may be very different between `priceAuthorities`.
+```js
+const myNotifier = E(pAuthority).makeQuoteNotifier(100quatloos, usdBrand);
```
Comment (katelynsills) [inline main/zoe/api/contract-support/price-authority.md:54]:
```suggestion
const myNotifier = E(pAuthority).makeQuoteNotifier(quatloos100, usdBrand);
```
can't start with a number

**Review by katelynsills — COMMENTED** [review]
Glanced at it, and here are some comments. I'll take another more detailed pass after Chris has a look

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
I haven't looked closely at the API descriptions yet.

File: main/zoe/api/contract-support/price-authority.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+of prices across the ecosystem, or might come directly from an AMM (Automatic
+Market Maker) like our 
+AutoSwap. A `PriceAuthority` can either give a quote for the current price 
+across any pair of currencies it knows about, or can immediately return a 
+`Promise` resolved when a condition is true. For example, a price 
+crossing some threshold, or at a particular time. It can also provide a 
+price feed that updates with every price change.
+
+A `PriceQuote` is an amount-payment pair, where the amount is also the current 
+balance of the payment:
```
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply main/zoe/api/contract-support/price-authority.md:17]: I think you're right. My first response seems to be a hold-over from Java, where `Pair` is a particular well-known type with problematic implications. I don't know whether it means anything specific to javascript developers. Interpreted as an extremely generic description, I don't have a problem with it.

File: main/zoe/api/contract-support/price-authority.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+A `PriceQuote` represents a statement from the PriceAuthority as to the 
+current price level at a particular time. The significant content (prices 
+and time) is packaged in the amount, and repeated in the payment for veracity. 
+A `PriceQuote` is an amount-payment pair, where the amount is also the current 
+balance of the payment.
+
+It represents a statement from the `PriceAuthority` as to the current price
+level at a particular time. The significant content (prices and time) is 
+packaged in the amount, and repeated in the payment for veracity.
```
Comment (Chris-Hibbert) [inline main/zoe/api/contract-support/price-authority.md:21]: "veracity" is redundant with the previous paragraph. It there anything here that isn't already present above or below?

File: main/zoe/api/contract-support/price-authority.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# PriceAuthority Methods
```
Comment (Chris-Hibbert) [inline main/zoe/api/contract-support/price-authority.md:1]: Just this morning I merged a PR that adds a bunch of new functionality that needs a separate discussion. The new methods are `mutableQuoteWhenLT,  mutableQuoteWhenLTE, mutableQuoteWhenGT`, and `mutableQuoteWhenGTE`.  The mutableQuoteWhen methods return a `mutableQuote` object with these methods: `cancel(e),   getPromise()`, and `updateLevel(newAmountIn, newAmountOutLimit)`.

`getPromise()` returns the same promise that you get from the `quoteWhen...()` methods, but additionally, you can call `updateLevel()` to change the trigger levels without requiring a second promise. This is designed to save effort and production of excessive objects (promises) for both the client of the priceAuthority and for the system.  Let me know if you have questions about the usage. `cancel()` causes the promise to `reject` with the message.

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]

File: main/zoe/api/contract-support/ratio-math.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+- `ratio`: `{ Ratio }`
+- Returns: `{ Amount }`
+
+Returns a hardened `Amount`.  Its brand is the `ratio`'s *numerator*'s brand.
+Its value is determined by:
+1. Multiplying the `amount` value by the `ratio`'s numerator's value.
+2. Dividing the result from step 1 by the `ratio`'s denominator's value.
+3. Applying a floor to the result from step 2 to round it down to
+   the nearest integer. If the step 2 result is already an integer, its value does
+   not change.
```
[the remark this answers is not in the document]
Comment (tyg) [reply main/zoe/api/contract-support/ratio-math.md:95]: Yep, I understood this. But I'm not trying to tell them exactly how the source code does it, but what the computation steps are; i.e. what are the separate things that happen to get to the result. So, yeah, 2 and 3 are implemented as a single floorDivide operation, but when explaining what's happening, I want to break it up into the two steps that take place in that single operation. To put it another way, I want them to know *what* happens, not precisely *how* it happens. I think this is easier to understand with the steps separated out.

File: main/zoe/api/contract-support/price-authority.md (hunk 5 above)
Comment (tyg) [reply main/zoe/api/contract-support/price-authority.md:1]: I've made a new issue for this and will start work on it, but I'd just as soon go ahead and merge this in to get ratio math in the system.

File: main/zoe/api/contract-support/ratio-math.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply main/zoe/api/contract-support/ratio-math.md:97]: My concern was that people who understand how JavaScript implements arithmetic would be led astray. Step two implies that there's an intermediate floating point result. 

I'll leave this to your judgement.

**Review by Chris-Hibbert — APPROVED** [review]

**Pull request merged**